server/src/main/java/com/google/sps/utils/ReportReader.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Description: File for reading in rows of CSV data
+ * Author: Kira Toal
+ * Date: June 24, 2020
+ */
+package com.google.sps.utils;
+
+import com.google.sps.utils.Ad;
+import com.google.sps.utils.AdRowProcessor;
+import java.io.*;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import org.apache.commons.io.IOUtils; 
+
+public class ReportReader {
+
+  private static final String CSV_FILE_PATH = "/output.csv";
+  private static final String COLLECTION = "testing"; 
+  private static final int START_ROW_INDEX = 1;  // first csv row that gets read in to DB (1 based)
+  private static final int END_ROW_INDEX = 4; // last csv row (exclusive)
+
+  public static void readCSV(InputStream csvFile) throws Exception {
+    try {    
+      BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(csvFile));      
+      String[] currentRow = {}; // array stores single row of CSV data (one advertisement)
+      // read header rows separately
+      String primaryHeader = bufferedReader.readLine();
+      String secondaryHeader = bufferedReader.readLine();
+      int rowIndex = 1; 
+      String row = ""; 
+      while((row = bufferedReader.readLine()) != null && (rowIndex < END_ROW_INDEX)) {
+        currentRow = row.split(",(?=(?:[^\"]*\"[^\"]*\")*[^\"]*$)", -1); 
```

Comment:
Add a short comment explaining the regex here

Reply:
Got it

Reply:
Again, I am unsure as to why this is not marked as outdated. There is a comment explaining regex in ReportReader. I am marking this as resolved to show that it's been addressed.